node/pkg/watchers/evm/watcher.go

```diff
@@ -331,7 +387,14 @@ func (w *Watcher) Run(parentCtx context.Context) error {
 							zap.Uint64("current_block", blockNumberU),
 							zap.Uint64("observed_block", blockNumber),
 						)
-						w.msgC <- msg
+						pubErr := w.publishIfSafe(msg, ctx, tx, receipt)
+						if pubErr != nil {
+							logger.Error("could not publish message: transfer verification failed",
+								zap.String("msgId", msg.MessageIDString()),
+								zap.String("txHash", msg.TxIDString()),
+								zap.Error(pubErr),
+							)
+						}
```

Comment:
This chunk of code is used 4 more times below. Should we put it inside `publishIfSafe` and then have that function not return anything?

Reply:
My preference is to maintain the return value so that the calling code has an idea of whether or not an error occurred. We could think about reducing code duplication by passing a `*logger` as an argument to that function and allowing it to print the error message, but then the calling code would need another log call if someone wants to log data that isn't visible to `publishIfSafe`, e.g. the `blockNumber` which exists for some instances in this file but not others.